codex-rs/mcp-server/src/codex_message_processor.rs

```diff
@@ -92,6 +98,68 @@ impl CodexMessageProcessor {
             ClientRequest::RemoveConversationListener { request_id, params } => {
                 self.remove_conversation_listener(request_id, params).await;
             }
+            ClientRequest::LoginChatGpt { request_id } => {
+                self.login_chatgpt(request_id).await;
+            }
+        }
+    }
+
+    async fn login_chatgpt(&mut self, request_id: RequestId) {
+        let config =
+            match Config::load_with_cli_overrides(Default::default(), ConfigOverrides::default()) {
+                Ok(cfg) => cfg,
+                Err(err) => {
+                    let error = JSONRPCErrorError {
+                        code: INTERNAL_ERROR_CODE,
+                        message: format!("error loading config for login: {err}"),
+                        data: None,
+                    };
+                    self.outgoing.send_error(request_id, error).await;
+                    return;
+                }
+            };
+
+        let opts = LoginServerOptions::new(config.codex_home.clone(), CLIENT_ID.to_string());
+        let outgoing = self.outgoing.clone();
+        match run_login_server(opts, None) {
+            Ok(server) => {
+                self.login_server = Some(server.clone());
+                tokio::spawn(async move {
+                    let result =
```

Comment:
Should this send a notification with the server URL to the client?

Also, it feels like this request should support cancellation?